python_modules/dagster-graphql/dagster_graphql/schema/schedules/schedules.py

```diff
@@ -145,33 +145,22 @@ def resolve_futureTicks(
         limit: Optional[int] = None,
         until: Optional[float] = None,
     ):
+        # Default cursor to current time
         cursor = cursor or time.time()
-
-        tick_times: list[float] = []
-        time_iter = self._remote_schedule.execution_time_iterator(cursor)
-
-        if until:
-            currentTime = None
-            while (not currentTime or currentTime < until) and (
-                limit is None or len(tick_times) < limit
-            ):
-                try:
-                    currentTime = next(time_iter).timestamp()
-                    if currentTime < until:
-                        tick_times.append(currentTime)
-                except StopIteration:
-                    break
-        else:
-            limit = limit or 10
-            for _ in range(limit):
-                tick_times.append(next(time_iter).timestamp())
-
-        schedule_selector = self._remote_schedule.schedule_selector
-        future_ticks = [
-            GrapheneDryRunInstigationTick(schedule_selector, tick_time) for tick_time in tick_times
-        ]
-
-        new_cursor = tick_times[-1] + 1 if tick_times else cursor
+        # Default limit to 10 ticks if no other bound is provided
+        limit = (limit or 10) if until is None else limit
+        
+        future_ticks: List[GrapheneDryRunInstigationTick] = []
+        try: 
+            for val in self._remote_schedule.execution_time_iterator(cursor):
+                # Add ticks before until timestamp and/or before limit is reached
+                if (until is None or val.timestamp() < until) and (limit is None or len(future_ticks) < limit):
+                    future_ticks.append(GrapheneDryRunInstigationTick(self._remote_schedule.schedule_selector, val.timestamp()))
+        except check.CheckError:
+            # If the schedule is not valid, we will return an empty list
+            pass
```

Comment:
I think we should add a logging.exception call if we do this so we know what the error was